### PR TITLE
Increase decryption failure tracker grace period

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes https://github.com/element-hq/element-android/issues/8766

## Type of change

- [ ] Feature
- [ ] Bugfix
- [x] Technical
- [ ] Other :

## Tests

- Step 1
- Step 2
- Step ...

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/element-hq/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/element-hq/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/analytics/DecryptionFailureTracker.kt [hunk 1]
```diff
@@ -40,8 +40,8 @@ private data class DecryptionFailure(
 )
 private typealias DetailedErrorName = Pair<String, Error.Name>
 
-private const val GRACE_PERIOD_MILLIS = 4_000
-private const val CHECK_INTERVAL = 2_000L
+private const val GRACE_PERIOD_MILLIS = 30_000
+private const val CHECK_INTERVAL = 40_000L
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/analytics/DecryptionFailureTracker.kt:44]: I quickly check the code to see that the change from `GRACE_PERIOD_MILLIS > CHECK_INTERVAL` to `GRACE_PERIOD_MILLIS < CHECK_INTERVAL` was OK, and I think it's fine this those 2 value are unrelated.

**Review by bmarty — APPROVED** [review]
LGTM, thanks. Maybe add a file for the changelog?

File: vector/src/main/java/im/vector/app/features/analytics/DecryptionFailureTracker.kt (hunk 1 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/analytics/DecryptionFailureTracker.kt:44]: Actually that's right. Should probably be a bit lower.
Because if currently the error happens just after a `checkFailures` (called on `CHECK_INTERVAL` tick`), then that error will be checked after 40s. So it will mean that the grace period is extended to 40s?
I'll put 10s so that it's better respecting the grace period. Worst case could extend to 39s

Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/analytics/DecryptionFailureTracker.kt]: I need some additional work, this code is not working as I thought. It's gracing events if you close the timeline, and this won't work properly with an extended grace period
